tests/test_datasets.py

```diff
@@ -747,5 +749,79 @@ def test_scanpaths_from_fixations(fixation_indices):
     compare_fixations(sub_fixations, new_sub_fixations, crop_length=True)
 
 
+@pytest.mark.parametrize('attribute_name , attribute_value, invert_match', [('dva',1,False), ('some_strings','n',True), ('some_strings','b',True)])
+def test_filter_stimuli_by_attribute(file_stimuli_with_attributes, fixation_trains, attribute_name, attribute_value, invert_match):
```

Comment:
These tests should go into `tests/test_filter_datasets.py`, because the functions are from `filter_datasets.py`. You can import `compare_fixations` etc from `test_datasets.py`, and you can also extend the fixturesin `filter_datasets.py` to have the additional attributes etc from `test_datasets.py`.

Reply:
I am not sure if I get you on the " extend the fixturesin filter_datasets.py to have the additional attributes etc from test_datasets.py" part?

Reply:
the fixtures `Stimuli` and `FixationTrains` in `test_datasets` come with attributes. The ones in `test_filter_datasets` don't have any attributes because I copied them over before I introduced attributes. But since you need the fixtures to have attributes to be able to apply the filter functions, you'll need to make sure that the fixtures in `test_filter_datasets` have attributes too.

Reply:
fixed